validation-test/Sema/type_checker_perf/fast/issue-62390.swift

```diff
@@ -0,0 +1,11 @@
+// RUN: %target-typecheck-verify-swift -solver-expression-time-threshold=1
+// REQUIRES: tools-release,no_asan
+
+// https://github.com/apple/swift/issues/62390
+// Compiler is extremely slow to compile simple closure
+
+func testFn<U>(_: ((Int, Int)) -> U) {}
+
+testFn { (a, _) in
+  return ((a <= a && a >= a) || (a <= a && a >= a)) // Ok
+}
```

Comment:
Is there any reason to turn this into a scale test over `||`?

Reply:
Probably not, the body of the closure was not to blame for the perf issue.

Reply:
Not really indeed, the issue is with how types are transferred into a closure.